parler_tts/modeling_parler_tts.py

```diff
@@ -1041,92 +1218,65 @@ def prepare_inputs_for_generation(
             if attention_mask is not None:
                 attention_mask = attention_mask.repeat((2, 1))
 
+            if prompt_hidden_states is not None:
+                prompt_hidden_states = torch.concatenate(
+                    [prompt_hidden_states, torch.zeros_like(prompt_hidden_states)], dim=0
+                )
+
+            if prompt_attention_mask is not None:
+                prompt_attention_mask = torch.concatenate(
+                    [prompt_attention_mask, torch.zeros_like(prompt_attention_mask)], dim=0
+                )
+
         if past_key_values is not None:
             input_ids = input_ids[:, -1:]
 
+            # we only want to use prompt signal in the 1st generation step but keeping the attention mask
+            prompt_hidden_states = None
+
         return {
             "input_ids": input_ids,
             "attention_mask": attention_mask,
             "encoder_hidden_states": encoder_hidden_states,
             "encoder_attention_mask": encoder_attention_mask,
+            "prompt_hidden_states": prompt_hidden_states,
+            "prompt_attention_mask": prompt_attention_mask,
             "head_mask": head_mask,
             "cross_attn_head_mask": cross_attn_head_mask,
             "past_key_values": past_key_values,
             "use_cache": use_cache,
         }
 
     # Ignore copy
-    def build_delay_pattern_mask(self, input_ids: torch.LongTensor, pad_token_id: int, max_length: int = None):
+    def build_delay_pattern_mask(
```

Comment:
Is there a reason we essentially make this a function instead of method?

Reply:
Yes indeed, I'm using it in a `Dataset.map` and this avoid hashing issues!

Reply:
I think you can do something hacky in your training script where you do:

```python
build_delay_pattern_mask = model.build_delay_pattern_mask

def preprocess(batch):
    # do some preprocessing with the pattern mask
    batch["labels"] = build_delay_pattern_mask(labels)
    return batch

dataset = dataset.map(preprocess)
```

But this is also a sufficient workaround as well